src/ratio.jl

```diff
@@ -0,0 +1,31 @@
+# Base's Rational type is too slow because of its penchant for calling gcd and div.
+# So we roll our own.
```

Comment:
It would be worth opening an issue on the main julia repo to track this issue – calling gcd and lcm all the time is pretty awful for performance. There's also the chronic overflow issue that Rational has, which seems like something that would need to be address to use the for any kind of real computation.